libc/src/time/baremetal/localtime_r.cpp

```diff
@@ -0,0 +1,23 @@
+//===-- Implementation of localtime_r for baremetal -----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/time/localtime_r.h"
+#include "src/time/time_utils.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(struct tm *, localtime_r,
+                   (const time_t *timer, struct tm *buf)) {
+  if (timer == nullptr) {
+    return nullptr;
+  }
```

Comment:
@michaelrj-google Something we should probably discuss at the next meeting is handling of `nullptr` arguments. The standard doesn't specify what should happen in that case and looking at other related functions like `gmtime` or `ctime` we're not consistent in how we handle it

Reply:
I think the direction we're currently moving is towards using the `LIBC_CRASH_ON_NULLPTR` macro. I'll add it to the list of topics as well

Reply:
@zimirza Can you use `LIBC_CRASH_ON_NULLPTR` here and elsewhere in this change?
```suggestion
  LIBC_CRASH_ON_NULLPTR(timer);
```